src/components/settings/DataManagement/useGlobalImportFileParser.ts

```diff
@@ -0,0 +1,132 @@
+import { logError } from '@/services/exceptions'
+import ErrorCodes from '@/services/exceptions/ErrorCodes'
+import { migrateAddedSafes } from '@/services/ls-migration/addedSafes'
+import { migrateAddressBook } from '@/services/ls-migration/addressBook'
+import { isChecksummedAddress } from '@/utils/addresses'
+import type { AddressBook, AddressBookState } from '@/store/addressBookSlice'
+import type { AddedSafesState } from '@/store/addedSafesSlice'
+import type { SafeAppsState } from '@/store/safeAppsSlice'
+import type { SettingsState } from '@/store/settingsSlice'
+
+import { useMemo } from 'react'
+
+export const enum SAFE_EXPORT_VERSION {
+  V1 = '1.0',
+  V2 = '2.0',
+}
+
+export enum ImportErrors {
+  INVALID_VERSION = 'The file is not a Safe export.',
+  INVALID_JSON_FORMAT = 'The JSON format is invalid.',
+  NO_IMPORT_DATA_FOUND = 'This file contains no importable data.',
+}
+
+const countEntries = (data: { [chainId: string]: { [address: string]: unknown } }) =>
+  Object.values(data).reduce<number>((count, entry) => count + Object.keys(entry).length, 0)
+
+export const _filterValidAbEntries = (ab?: AddressBookState): AddressBookState | undefined => {
+  if (!ab) {
+    return undefined
+  }
+
+  return Object.entries(ab).reduce<AddressBookState>((acc, [chainId, chainAb]) => {
+    const sanitizedChainAb = Object.entries(chainAb).reduce<AddressBook>((acc, [address, name]) => {
+      if (name?.trim() && address && isChecksummedAddress(address)) {
```

Comment:
Is it possible for `name` or `address` to be undefined? If so we should leave a comment why it is not reflected in the type.